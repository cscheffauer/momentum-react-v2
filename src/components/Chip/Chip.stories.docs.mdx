The `<Chip />` component.

Rendering LabelChip from momentum-design