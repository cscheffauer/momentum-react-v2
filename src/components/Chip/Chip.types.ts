import type { Chip } from '@momentum-design/components/dist/react';

// props type of chip
type MdcChipProps = React.ComponentProps<typeof Chip>;

export type Props = MdcChipProps & {
  stopPropagation?: boolean;
};
