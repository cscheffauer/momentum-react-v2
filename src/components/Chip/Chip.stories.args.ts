import { commonStyles } from '../../storybook/helper.stories.argtypes';
import {} from '@momentum-design/components';
const chipArgTypes = {
  label: {
    description: 'The visible label text of the chip',
    control: { type: 'text' },
    table: {
      type: {
        summary: 'string',
      },
      defaultValue: {
        summary: 'undefined',
      },
    },
  },
  iconName: {
    description:
      'The name of the icon in the chip (must match the icon name in the Icon component)',
    control: { type: 'text' },
    table: {
      type: {
        summary: 'string',
      },
      defaultValue: {
        summary: 'undefined',
      },
    },
  },
  color: {
    description: 'Sets the chip color',
    control: { type: 'select' },
    options: [
      undefined,
      'default',
      'cobalt',
      'gold',
      'lime',
      'mint',
      'pink',
      'purple',
      'slate',
      'violet',
    ],
    table: {
      type: {
        summary: 'Color',
      },
      defaultValue: {
        summary: 'default',
      },
    },
  },
  disabled: {
    description: 'Sets whether the chip is disabled',
    control: { type: 'boolean' },
    table: {
      type: {
        summary: 'boolean',
      },
      defaultValue: {
        summary: false,
      },
    },
  },
  onClick: {
    action: 'onClick',
    description: 'Handler that is called when the element is clicked',
    table: {
      category: 'Mdc Button Attributes',
      type: {
        summary: '(e: MouseEvent) => void',
      },
      defaultValue: {
        summary: 'undefined',
      },
    },
  },
  onFocus: {
    action: 'onFocus',
    description: 'Handler that is called when the element receives focus.',
    table: {
      category: 'Mdc Button Attributes',
      type: {
        summary: '(e: FocusEvent) => void',
      },
      defaultValue: {
        summary: 'undefined',
      },
    },
  },
  onBlur: {
    action: 'onBlur',
    description: 'Handler that is called when the element loses focus.',
    table: {
      category: 'Mdc Button Attributes',
      type: {
        summary: '(e: FocusEvent) => void',
      },
      defaultValue: {
        summary: 'undefined',
      },
    },
  },
  onKeyDown: {
    action: 'onKeyDown',
    description: 'Handler that is called when a key is pressed.',
    table: {
      category: 'Mdc Button Attributes',
      type: {
        summary: '(e: KeyboardEvent) => void',
      },
      defaultValue: {
        summary: 'undefined',
      },
    },
  },
  onKeyUp: {
    action: 'onKeyUp',
    description: 'Handler that is called when a key is released.',
    table: {
      category: 'Mdc Button Attributes',
      type: {
        summary: '(e: KeyboardEvent) => void',
      },
      defaultValue: {
        summary: 'undefined',
      },
    },
  },
};

export { chipArgTypes };

export default {
  ...commonStyles,
  ...chipArgTypes,
};
