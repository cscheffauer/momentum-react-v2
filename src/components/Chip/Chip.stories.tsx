import { Template } from '../../storybook/helper.stories.templates';
import { DocumentationPage } from '../../storybook/helper.stories.docs';
import StyleDocs from '../../storybook/docs.stories.style.mdx';

import Chip, { ChipProps } from './';
import argTypes from './Chip.stories.args';
import Documentation from './Chip.stories.docs.mdx';

export default {
  title: 'Momentum UI/Chip',
  component: Chip,
  parameters: {
    expanded: true,
    docs: {
      page: DocumentationPage(Documentation, StyleDocs),
    },
  },
};

const Example = Template<ChipProps>(Chip).bind({});

Example.argTypes = { ...argTypes };

Example.args = {
  label: 'Example',
  color: 'default',
};

export { Example };
