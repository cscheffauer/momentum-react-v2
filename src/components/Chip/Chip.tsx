import React, { FC } from 'react';

import { Props } from './Chip.types';

import { Chip as MdcChip } from '@momentum-design/components/dist/react';

/**
 * The Chip component.
 */
const Chip: FC<Props> = (props: Props) => {
  return <MdcChip {...props} />;
};

export default Chip;
