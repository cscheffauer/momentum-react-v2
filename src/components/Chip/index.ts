import { default as Chip } from './Chip';
import { Props } from './Chip.types';

export type ChipProps = Props;

export default Chip;
