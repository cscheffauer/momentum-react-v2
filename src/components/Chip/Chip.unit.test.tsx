import React from 'react';
import { mount } from 'enzyme';

import Chip from './';

const label = 'Some chip text';

describe('<Chip />', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      expect.assertions(1);

      const container = mount(<Chip />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with className', () => {
      expect.assertions(1);

      const className = 'example-class';

      const container = mount(<Chip className={className} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with id', () => {
      expect.assertions(1);

      const id = 'example-id';

      const container = mount(<Chip id={id} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with style', () => {
      expect.assertions(1);

      const style = { color: 'pink' };

      const container = mount(<Chip style={style} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with text', () => {
      const container = mount(<Chip label={label} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with icon', () => {
      const container = mount(<Chip iconName={'accessibility-regular'} />);

      expect(container).toMatchSnapshot();
    });

    it('it should match snapshot with disabled', () => {
      const container = mount(<Chip disabled />);

      expect(container).toMatchSnapshot();
    });
  });
});
