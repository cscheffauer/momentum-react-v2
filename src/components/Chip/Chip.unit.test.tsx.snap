// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chip /> snapshot it should match snapshot with disabled 1`] = `
<Chip
  disabled={true}
>
  <Chip
    disabled={true}
  >
    <mdc-chip
      suppressHydrationWarning={true}
    />
  </Chip>
</Chip>
`;

exports[`<Chip /> snapshot should match snapshot 1`] = `
<Chip>
  <Chip>
    <mdc-chip
      suppressHydrationWarning={true}
    />
  </Chip>
</Chip>
`;

exports[`<Chip /> snapshot should match snapshot with className 1`] = `
<Chip
  className="example-class"
>
  <Chip
    className="example-class"
  >
    <mdc-chip
      class="example-class"
      suppressHydrationWarning={true}
    />
  </Chip>
</Chip>
`;

exports[`<Chip /> snapshot should match snapshot with icon 1`] = `
<Chip
  iconName="accessibility-regular"
>
  <Chip
    iconName="accessibility-regular"
  >
    <mdc-chip
      suppressHydrationWarning={true}
    />
  </Chip>
</Chip>
`;

exports[`<Chip /> snapshot should match snapshot with id 1`] = `
<Chip
  id="example-id"
>
  <Chip
    id="example-id"
  >
    <mdc-chip
      suppressHydrationWarning={true}
    />
  </Chip>
</Chip>
`;

exports[`<Chip /> snapshot should match snapshot with style 1`] = `
<Chip
  style={
    Object {
      "color": "pink",
    }
  }
>
  <Chip
    style={
      Object {
        "color": "pink",
      }
    }
  >
    <mdc-chip
      style={
        Object {
          "color": "pink",
        }
      }
      suppressHydrationWarning={true}
    />
  </Chip>
</Chip>
`;

exports[`<Chip /> snapshot should match snapshot with text 1`] = `
<Chip
  label="Some chip text"
>
  <Chip
    label="Some chip text"
  >
    <mdc-chip
      suppressHydrationWarning={true}
    />
  </Chip>
</Chip>
`;
